extensions/arc/src/extension.ts

```diff
@@ -5,102 +5,41 @@
 import * as loc from './localizedConstants';
-import { IconPathHelper } from './constants';
-import { BasicAuth } from './controller/auth';
-import { PostgresDashboard } from './ui/dashboards/postgres/postgresDashboard';
-import { ControllerModel } from './models/controllerModel';
-import { PostgresModel } from './models/postgresModel';
-import { ControllerDashboard } from './ui/dashboards/controller/controllerDashboard';
-import { MiaaDashboard } from './ui/dashboards/miaa/miaaDashboard';
-import { MiaaModel } from './models/miaaModel';
-import { AzureArcTreeDataProvider } from './ui/tree/controllerTreeDataProvider';
+import { IconPathHelper, refreshActionId } from './constants';
+import { AzureArcTreeDataProvider } from './ui/tree/azureArcTreeDataProvider';
 import { ControllerTreeNode } from './ui/tree/controllerTreeNode';
 import { TreeNode } from './ui/tree/treeNode';
+import { ConnectToControllerDialog } from './ui/dialogs/connectControllerDialog';
 
 export async function activate(context: vscode.ExtensionContext): Promise<void> {
 	IconPathHelper.setExtensionContext(context);
 
 	const treeDataProvider = new AzureArcTreeDataProvider(context);
 	vscode.window.registerTreeDataProvider('azureArc', treeDataProvider);
 
-	vscode.commands.registerCommand('arc.addController', () => {
-		// Controller information
-		const controllerUrl = '';
-		const auth = new BasicAuth('', '');
-
-		const controllerModel = new ControllerModel(controllerUrl, auth);
-		treeDataProvider.addController(controllerModel);
-	});
-
-	vscode.commands.registerCommand('arc.removeController', (controllerNode: ControllerTreeNode) => {
-		treeDataProvider.removeController(controllerNode);
+	vscode.commands.registerCommand('arc.createController', async () => {
+		await vscode.commands.executeCommand('azdata.resource.deploy');
 	});
 
-	vscode.commands.registerCommand('arc.openDashboard', async (treeNode: TreeNode) => {
-		await treeNode.openDashboard().catch(err => vscode.window.showErrorMessage(loc.openDashboardFailed(err)));
-	});
-
-	vscode.commands.registerCommand('arc.manageArcController', async () => {
-		// Controller information
-		const controllerUrl = '';
-		const auth = new BasicAuth('', '');
-
-		try {
-			const controllerModel = new ControllerModel(controllerUrl, auth);
-			const controllerDashboard = new ControllerDashboard(controllerModel);
-
-			await Promise.all([
-				controllerDashboard.showDashboard(),
-				controllerModel.refresh()
-			]);
-		} catch (error) {
-			// vscode.window.showErrorMessage(loc.failedToManagePostgres(`${dbNamespace}.${dbName}`, error));
+	vscode.commands.registerCommand('arc.connectToController', async () => {
+		const dialog = new ConnectToControllerDialog(treeDataProvider);
+		dialog.showDialog();
+		const model = await dialog.waitForClose();
+		if (model) {
+			await treeDataProvider.addOrUpdateController(model.controllerModel, model.password);
 		}
 	});
 
-	vscode.commands.registerCommand('arc.manageMiaa', async () => {
```

Comment:
Why are some commands no longer registered here?

Reply:
Because those were just temporary and aren't needed anymore. The general "manage" command handles launching the dashboards now.